File: internal/models/book.go
package models

import "github.com/Goldmite/project_go/internal/enums"

type Book struct {
	ISBN        string         `json:"isbn"`
	Title       string         `json:"title"`
	Authors     []string       `json:"authors"`
	Pages       uint           `json:"pageCount"`
	Description string         `json:"description"`
	Publisher   string         `json:"publisher"`
	PublishDate string         `json:"publishedDate"`
	Language    enums.Language `json:"language"`
	Cover       Image          `json:"imageLinks"`
}

type Image struct {
	Url string `json:"thumbnail"`
}

type BookItem struct {
	Book Book `json:"volumeInfo"`
}

type BookQueryResponse struct {
	Items []BookItem `json:"items"`
}

File: frontend/src/lib/types/responses/book.ts
export interface Book {
	owned_by: string[];
	isbn: string;
	title: string;
	authors: string[];
	pages: number;
	description: string;
	publisher: string;
	publishedDate: string;
	language: string;
	cover: string;
}

export interface BookProgress {
	pages_read: number;
	time_read: number;
	first_page: number;
	current_page: number;
}

export interface RecentlyRead {
	isbn: string;
	title: string;
	authors: string[];
	cover: string;
	start_date: string;
	last_read_date: string;
}
